src/Cardano/Wallet.hs

```diff
@@ -118,36 +119,46 @@ data NewWallet = NewWallet
         :: !WalletName
     , passphrase
         :: !(Passphrase "encryption")
-    , gap
-        :: !AddressPoolGap
-    } deriving (Show, Generic)
+    , addressDiscoveryConfig
+        :: AddressDiscoveryConfig s
+    }
```

Comment:
A suggestion here: let's remove the `NewWallet` type wrapper here and simply modify the wallet layer signature as follows:

```hs
        :: WalletName
        -> Passphrase "seed"
        -> Passphrase "generation"
        -> Passphrase "encryption" 
        -> s
        -> ExceptT (ErrWalletAlreadyExists "createWallet") IO WalletId
```

Leave the creation of the state to the caller :+1:

Reply:
OK good idea. Turns out `NewWallet` is kind of helpful, so I have added a `createWalletSeq :: WalletLayer SeqState -> NewWallet ExceptT ErrWalletAlreadyExists IO WalletId`.